Create UI for add-maintainer

jnnybk opened the pull request:
(NOTE) This PR should rebase master once PR #44 is merged.

This PR accomplishes:
1) Adding "Add Maintainer" option to the NavBar for Maintainers.
2) Creates Pop-Up to input User Email to send to /add-maintainer servlet.
![Screenshot 2020-07-15 at 2 47 46 PM - Display 2](https://user-images.githubusercontent.com/64429989/87583662-502ba900-c6aa-11ea-84b2-7d688fdca721.png)

jnnybk committed:
Add maintainerForm to navbar

jnnybk committed:
Add styling for pop up

jnnybk committed:
Fix logic error in AddMaintainer

jnnybk committed:
Fix styling for Add-Maintainer UI

jnnybk committed:
Add new-maintainer-email label

Sarah-Addo reviewed (commented):
could you have the popup overlay the page like https://screenshot.googleplex.com/JJNuVhuxt6C?

jnnybk committed:
Merge branch 'master' into implement-ismaintainer-check

jnnybk committed:
add styling

jnnybk committed:
Add styling to fix pop up

jnnybk commented:
Updates to this PR's completed tasks:
1) Created a modal for adding maintainer. 
2) Fix logic error in AddMaintainerServlet.java
3) Removed maintainer.js, added it to user.js when building the navBar

![Screenshot 2020-07-15 at 7 30 26 PM - Display 2](https://user-images.githubusercontent.com/64429989/87609857-b2e46b00-c6d1-11ea-8706-7d2f60df745a.png)

jnnybk committed:
Merge branch 'master' into implement-ismaintainer-check

jnnybk committed:
Add redirect to index.html once Maintainer is successfully added

jnnybk committed:
Merge branch 'implement-ismaintainer-check' of https://github.com/googleinterns/step130-2020 into implement-ismaintainer-check

jnnybk committed:
Fix styling and whitespace

jnnybk committed:
Add null check

belraquib commented on src/main/java/com/google/sps/servlets/AddMaintainerServlet.java:
```diff
@@ -53,13 +53,14 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     } else {
       addNewMaintainerToDatastore(newMaintainerEmail);
     }
     response.setStatus(HttpServletResponse.SC_OK);
```
You don't want to set OK and sendRedirect. Pick one.

belraquib commented on src/main/webapp/user.js:
```diff
@@ -58,11 +91,30 @@ class User {
+        addMaintainerPopupBackground.classList.add("add-maintainer-popup-modal-background");
+        addMaintainerPopupBackground.appendChild(addMaintainerPopup);
+
+        const mainContent = document.getElementById("help-near-me-content") ? document.getElementById("help-near-me-content") : document.getElementById("organizations-content");
```
This seems weird to me. Why add it to the mainContent and not document.body ? As is, you are leaking what page is hosting this popup into the logic here. What happens when/if you have a new page?

jnnybk replied:
I agree! I fixed it to prepend to the body of the document instead of to specific pages.

jnnybk committed:
Remove setting response status

jnnybk committed:
Prepend to body instead of specifying individual pages

jnnybk committed:
Merge branch 'master' into implement-ismaintainer-check
